shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profilePreferences/data/DbUserDataSource.kt

```diff
@@ -0,0 +1,62 @@
+package com.vml.tutorial.plantshop.profilePreferences.data
+
+import com.vml.tutorial.plantshop.PlantDatabase
+import com.vml.tutorial.plantshop.core.utils.exts.orZero
+import com.vml.tutorial.plantshop.profilePreferences.domain.Address
+import com.vml.tutorial.plantshop.profilePreferences.domain.User
+import com.vml.tutorial.plantshop.profilePreferences.domain.UserDataSource
+import database.UserQueries
+
+class DbUserDataSource(db: PlantDatabase) : UserDataSource {
+    private val queries: UserQueries = db.userQueries
+    override suspend fun insertToDatabase(user: User) {
+        queries.insertUser(
+            cId = user.cId,
+            firstName = user.firstName.orEmpty(),
+            lastName = user.lastName.orEmpty(),
+            email = user.email,
+            birthDate = user.birthDate.orEmpty(),
+            phoneNumber = user.phoneNumber.orEmpty(),
+            streetName = user.address.streetName.orEmpty(),
```

Comment:
Maybe it is just my problem and it is not gonna happen with the new users, but can you make the address field nullable? For me, the app is crashing right now, because it doesn't exist.

Reply:
Fixed